ivy/functional/backends/jax/device.py

```diff
@@ -50,53 +57,70 @@ def dev(
 
 def to_device(
     x: JaxArray,
-    device: jaxlib.xla_extension.Device,
+    device: Union[jaxlib.xla_extension.Device, ivy.Device],
     /,
     *,
     stream: Optional[int] = None,
     out: Optional[JaxArray] = None,
 ):
-    if device is not None:
-        cur_dev = as_native_dev(dev(x))
-        if cur_dev != device:
-            x = jax.device_put(x, as_native_dev(device))
-    return x
+    # TODO: implement stream and out
+    return _to_device(x, device=device)
 
 
 # this is a non-wrapped function used to place JAX arrays on respective devices,
 # since if we use to_device, it will return ivy.array which is not desirable
 def _to_device(x, device=None):
     if device is not None:
-        cur_dev = as_native_dev(dev(x))
+        device = as_native_dev(device)
+        cur_dev = dev(x, as_native=True)
         if cur_dev != device:
-            x = jax.device_put(x, as_native_dev(device))
+            # Uses async dispatch, data commited only when required
+            x = jax.device_put(x, device)
     return x
```

Comment:
this function is redundant as calling `to_device` directly should return native arrays if called correctly, I suggest removing it and using the native `to_device`